CRM/Core/Form.php

```diff
@@ -364,11 +364,19 @@ public function registerRules() {
    */
   public function &add(
     $type, $name, $label = '',
-    $attributes = '', $required = FALSE, $extra = NULL
+    $attributes = NULL, $required = FALSE, $extra = NULL
   ) {
     if ($type === 'radio') {
       CRM_Core_Error::deprecatedFunctionWarning('CRM_Core_Form::addRadio');
     }
+
+    if ($attributes && !is_array($attributes)) {
+      // The $attributes param used to allow for strings and would default to an
+      // empty string.  However, now that the variable is heavily manipulated,
+      // we should expect it to always be an array.  However, an empty array can
+      // cause problems.
```

Comment:
Why would an empty array cause problems? You mean it's not a good default value? I agree that NULL seems like the sensible choice as the default for that param in this function. But when I first read this comment it sounded like you were saying it would cause problems if you called this function and passed an empty array. I don't think that's the case right?

Reply:
It has been a few months since I wrote that, so I'm not certain why.  However, I do know that I tried a bunch of possible approaches, and I think I tried making the param default to an empty array rather than `null` at one point and found problems.

As you can probably imagine from reading that function and from seeing it used all over the place, it receives a mess of things depending upon the context and `$type`.  Some of this was resolved through identifying and fixing things through the deprecation warning (one of the later commits), so it might not even be a problem now.

I can edit the comment if you think it's misleading.

Reply:
Understood. I think `NULL` is a sensible default for the function param. I would :+1: for updating the comment.